public/app/core/utils/query.ts

```diff
@@ -1,5 +1,6 @@
 import { DataQuery, DataSourceRef } from '@grafana/data';
 
+// Legacy. Import from packages/grafana-data
```

Comment:
```suggestion
// @deprecared use the `getNextRefId` function from grafana/data instead
```
I think if you change the comment to this, the code editors (at least the IntelliJ-based ones) will mark this function as deprecated.

Reply:
Done, thanks @Clarity-89!